MediaBrowser.Providers/MediaInfo/AudioResolver.cs

```diff
@@ -0,0 +1,164 @@
+            bool clearCache,
+            ILocalizationManager localizationManager,
+            IMediaEncoder mediaEncoder,
+            CancellationToken cancellationToken)
+        {
+            var streams = new List<MediaStream>();
+
+            if (!video.IsFileProtocol)
+            {
+                return streams;
+            }
+
+            List<string> paths = GetExternalAudioFiles(video, directoryService, namingOptions, clearCache);
+
+            await AddExternalAudioStreams(streams, paths, startIndex, localizationManager, mediaEncoder, cancellationToken);
+
+            return streams;
+        }
+
+        public List<string> GetExternalAudioFiles(
+            Video video,
+            IDirectoryService directoryService,
+            NamingOptions namingOptions,
+            bool clearCache)
+        {
+            List<string> paths = new List<string>();
+
+            if (!video.IsFileProtocol)
+            {
+                return paths;
+            }
+
+            paths.AddRange(GetAudioFilesFromFolder(video.ContainingFolderPath, video.Path, directoryService, namingOptions, clearCache));
+            paths.AddRange(GetAudioFilesFromFolder(video.GetInternalMetadataPath(), video.Path, directoryService, namingOptions, clearCache));
+
+            return paths;
+        }
+
+        private List<string> GetAudioFilesFromFolder(
+            string folder,
+            string videoFileName,
+            IDirectoryService directoryService,
+            NamingOptions namingOptions,
+            bool clearCache)
+        {
+            List<string> paths = new List<string>();
+            string videoFileNameWithoutExtension = Path.GetFileNameWithoutExtension(videoFileName);
+
+            if (!Directory.Exists(folder))
+            {
+                return paths;
+            }
+
+            var files = directoryService.GetFilePaths(folder, clearCache, true);
+            for (int i = 0; i < files.Count; i++)
+            {
+                string fileNameWithoutExtension = Path.GetFileNameWithoutExtension(files[i]);
+
+                if (!AudioFileParser.IsAudioFile(files[i], namingOptions))
+                {
+                    continue;
+                }
+
+                // The audio filename must either be equal to the video filename or start with the video filename followed by a dot
+                if (videoFileNameWithoutExtension.Equals(fileNameWithoutExtension, StringComparison.OrdinalIgnoreCase) ||
+                    (fileNameWithoutExtension.Length > videoFileNameWithoutExtension.Length
+                         && fileNameWithoutExtension[videoFileNameWithoutExtension.Length] == '.'
+                         && fileNameWithoutExtension.StartsWith(videoFileNameWithoutExtension, StringComparison.OrdinalIgnoreCase)))
+                {
+                    paths.Add(files[i]);
+                }
+            }
+
+            return paths;
+        }
+
+        public async Task AddExternalAudioStreams(
```

Comment:
Afaict, this method is only used once in this class. You could just move the body of the method into the calling method `GetExternalAudioStreams`.